### ISSUE TITLE
DOTNET_SHUTDOWNTIMEOUTSECONDS not respected

### ISSUE THREAD

**Issue by LarsAndreasEk** [dotnet/extensions#2942]
### Setting ShutdownTimeout via environment variable does not affect host shutdown timeout
According to the [docs](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/generic-host?view=aspnetcore-3.1#shutdowntimeout), setting the DOTNET_SHUTDOWNTIMEOUTSECONDS as an environment variable should affect graceful shutdown time. Configuring it through code work fine.

### To Reproduce
1. Create new api project from VS template
2. Set the environment variable in debug settings (or otherwise), to a value other than the 5 seconds default.
3. Add an IOptions< HostOptions > dependency to the default created controller and return the ShutdownTimeout.
4. Start in debug.
5. In the browser, see that the value is still 5 seconds.

### Expected behavior
The HostOptions.ShutdownTimeout value should be set to whatever value passed to it by DOTNET_SHUTDOWNTIMEOUTSECONDS.

Tested both with 3.1 and 3.0.1.

Please see attached repro.
[ReproduceShutdownTimeout.zip](https://github.com/dotnet/extensions/files/4164843/ReproduceShutdownTimeout.zip)

**Comment by Pilchie** [comment]
@anurse should this move to aspnetcore and servers?

**Comment by anurse** [comment]
@Pilchie no, this is in Hosting (and Generic Host, to be specific) so this repo is the right place.

**Comment by anurse** [comment]
@LarsAndreasEk looking at the code, I'm not sure where `SHUTDOWNTIMEOUTSECONDS` came from. It's possible the docs are out of date. What happens if you set `DOTNET_SHUTDOWNTIMEOUT` to a value that's [parsable as a `System.TimeSpan`](https://docs.microsoft.com/en-us/dotnet/api/system.timespan.parse?view=netframework-4.8#System_TimeSpan_Parse_System_String_)? For example, for 10 seconds: `00:00:10`.

If that works, I think it's probably best we update the docs, since I think using a `TimeSpan` is more flexible here.

**Comment by LarsAndreasEk** [comment]
I’m relatively sure I tested that too. Can double check when I’m back at
work tomorrow.

**Comment by LarsAndreasEk** [comment]
@anurse just tested setting DOTNET_SHUTDOWNTIMEOUT to 00:00:10 and found it not to work.

**Comment by jkotalik** [comment]
Let's investigate if the option works in general (either with ASPNETCORE_ or DOTNET_ prefix).
